cnd/src/network.rs

```diff
@@ -868,130 +843,38 @@ impl libp2p::swarm::NetworkBehaviourEventProcess<()> for ComitNode {
-
-                let halight_params = halight::Params {
-                    redeem_identity: halight_redeem_identity,
-                    refund_identity: halight_refund_identity,
-                    cltv_expiry: create_swap_params.lightning_cltv_expiry,
-                    asset: create_swap_params.lightning_amount,
-                    secret_hash,
-                };
-
-                let halight_states = self.halight_states.clone();
-                let halight_watcher_task = match role {
-                    Role::Alice => async move {
-                        halight::new(
-                            local_swap_id,
-                            halight_params,
-                            Role::Alice,
-                            Side::Beta,
-                            halight_states,
-                            LndConnectorAsReceiver::from(lnd_connector_params),
-                        )
-                        .await;
-                    }
-                    .boxed(),
-                    Role::Bob => async move {
-                        halight::new(
-                            local_swap_id,
-                            halight_params,
-                            Role::Bob,
-                            Side::Beta,
-                            halight_states,
-                            LndConnectorAsSender::from(lnd_connector_params),
-                        )
-                        .await;
-                    }
-                    .boxed(),
-                };
-                self.task_executor.spawn(halight_watcher_task);
-
-                // third, we spawn the watcher for herc20
-                let (herc20_redeem_identity, herc20_refund_identity) = match role {
-                    Role::Alice => (ethereum_identity, create_swap_params.ethereum_identity),
-                    Role::Bob => (create_swap_params.ethereum_identity, ethereum_identity),
-                };
-                let params = herc20::Params {
-                    asset: Erc20::new(
-                        create_swap_params.token_contract,
-                        create_swap_params.ethereum_amount,
-                    ),
-                    redeem_identity: herc20_redeem_identity,
-                    refund_identity: herc20_refund_identity,
-                    expiry: create_swap_params.ethereum_absolute_expiry,
-                    secret_hash,
-                };
-
-                self.task_executor.spawn(herc20::new(
-                    local_swap_id,
-                    params,
-                    Utc::now().naive_local(),
-                    role,
-                    Side::Alpha,
-                    self.herc20_states.clone(),
-                    self.ethereum_connector.clone(),
-                ));
+        let comit::BehaviourOutEvent {
+            local_swap_id,
+            remote_data,
+        } = event;
+
+        let storage = self.storage.clone();
+        // FIXME: These are in  storage but not accessible?
```

Comment:
Help is on the way: https://github.com/comit-network/comit-rs/blob/respawn-swaps/cnd/src/protocol_spawner.rs

Reply:
Link is broken

Reply:
Oh my, that is nice. I have used it in this PR now :)